File: frontend/lib/features/linklist/domain/entities/LinkEntity.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'LinkEntity.freezed.dart';
part 'LinkEntity.g.dart';

@freezed
class LinkEntity with _$LinkEntity {
  const factory LinkEntity({
    required int id,
    required int category,
    required String title,
    required String url,
    required String description,
    required bool support,
    required String contact,
    required String image,
    required int likes,
    required int dislikes,
    required bool active,
    required String categoryTitle,
  }) = _LinkedEntity;

  factory LinkEntity.fromJson(Map<String, Object?> json) => _$LinkEntityFromJson(json);
}

File: src/main/java/cool/cfapps/reactivelinkappxplatform/entity/LinkEntity.java
package cool.cfapps.reactivelinkappxplatform.entity;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.Id;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class LinkEntity {
    @Id
    private long id;
    private long category;
    private String title;
    private String url;
    private String description;
    private boolean support;
    private String contact;
    private String image;
    private int likes;
    private int dislikes;
    private boolean active;
}
